### ISSUE TITLE
Crash deleting CircleArena.floorAppearance node

### ISSUE THREAD

**Issue by stefaniapedrazzi** [cyberbotics/webots#1869]
Webots often crashes when deleting the `CircleArena.floorAppearance` node.
This field is a template regenerator field, thus deleting the nodes triggers the complete regeneration (deletion and new creation) of the CircleArena PROTO node.
The Supervisor API requires to be notified about the deleted nodes, but it seems that the deleted nodes info links to an invalid `parentField` (deleted during the PROTO regeneration).

Steps to reproduce:
1. Open the `projects/languages/cpp/worlds/example.wbt`
2. Run the simulation
3. Delete `CircleArena.floorAppearance`

Stack trace:
```
Thread 1 "webots-bin" received signal SIGSEGV, Segmentation fault.
0x00007ffff7728f01 in ?? ()
(gdb) bt
#0  0x00007ffff7728f01 in ?? ()
#1  0x0000555562156110 in ?? ()
#2  0x0000555555902408 in QString::toUtf8() const & (this=0x0) at ../../include/qt/QtCore/QtCore/qstring.h:679
#3  0x0000555555b4a745 in WbSupervisorUtilities::writeAnswer (this=0x5555569f5e90, stream=...) at nodes/utils/WbSupervisorUtilities.cpp:1309
#4  0x0000555555acf633 in WbRobot::writeAnswer (this=0x55555c26da20, stream=...) at nodes/WbRobot.cpp:1177
#5  0x0000555555ace5fa in WbRobot::dispatchAnswer (this=0x55555c26da20, stream=..., includeDevices=true) at nodes/WbRobot.cpp:1032
#6  0x0000555555978109 in WbController::writeAnswer (this=0x555564976c60, immediateAnswer=false) at control/WbController.cpp:924
#7  0x000055555596d7fd in WbControlledWorld::waitForRobotWindowIfNeededAndCompleteStep (this=0x5555638413e0) at control/WbControlledWorld.cpp:478
#8  0x000055555596c941 in WbControlledWorld::step (this=0x5555638413e0) at control/WbControlledWorld.cpp:355
#9  0x000055555596c181 in WbControlledWorld::processWaitingStep (this=0x5555638413e0) at control/WbControlledWorld.cpp:268
#10 0x000055555596c13f in WbControlledWorld::triggerStepFromTimer (this=0x5555638413e0) at control/WbControlledWorld.cpp:262
```

**Comment by stefaniapedrazzi** [comment]
Should also be fixed in #1868

**Comment by stefaniapedrazzi** [comment]
No, I checked again and this is not fixed.
But could be related to #1885.

**Comment by stefaniapedrazzi** [comment]
I will fix this in #1868.
